### ISSUE TITLE
Amazon Fire TV Stick and Mouse Issue

### ISSUE THREAD

**Issue by DaShermanator** [moonlight-stream/moonlight-android#1130]
When playing any FPS game (Overwatch, CSGO, COD ect) my mouse does not function correctly. The crosshairs are stuck at the center of screen with the perspective of either looking up at the sky or down at the floor and mouse movement does nothing but spin the character in circles on the Z axis. 

Moonlight running on Fire TV Stick with mouse and keyboard connected to the Fire TV Stick. I have tried both bluetooth/2.4 dongle for wireless connection to the Fire TV Stick or wired connection using a USB hub. Both yield the same result.

My 8BitDo controller works great. However, I have no chance competing with keyboard and mouse users in games like Overwatch or CSGO. I think it's important to note that other games, non-FPS games (strategy, top-down, 2-D) work fine with keyboard and mouse. It seems as though the Firestick does not recognize the mouse input for FPS games however, I cannot find anything online regarding this issue.

I am turning my office into a nursery and have to game somewhere else so I was hoping to get this working for playability on other screens in my apartment. Any solution, fix or advice (even on a different setup) would be appreciated. Keep in mind I'm working on a budget. 

Thank you. 
-Chris

**Comment by cgutman** [comment]
Which model of the Fire TV Stick are you using?

If your Fire TV Stick is an older model, it may not have a new enough version of Fire OS to support proper mouse capture. It would need Fire OS 7 or later (based on Android 9) to support this, since the feature was introduced originally in Android 8.0.

**Comment by DaShermanator** [comment]
Thank you for the prompt reply cgutman. 

My Fire TV stick is 3rd Generation (Model Number: S3L46N) purchased October 2021. I verified it is running Fire OS 7.2. 

I would think the fire stick is a common device for moonlight given it's the most economical ($25 on Prime day). I'm surprised I can't find anything about this online. 

I would appreciate any other suggestions. Can anyone out there duplicate this issue?

**Comment by cgutman** [comment]
Do you by chance have the "Remote desktop mouse mode" option enabled? That can cause issues with FPS games.

**Comment by DaShermanator** [comment]
No. That was one of my first attempted solution. Toggling "Remote Desktop Mouse Mode" made no discernible difference.

**Comment by DaShermanator** [comment]
Should I try side loading Moonlight on the fire stick?

**Comment by cgutman** [comment]
The APK version posted here is the same as what's on the Amazon Appstore with the exception of Amazon's code signing.

I tried to reproduce this bug by pairing a Bluetooth mouse to my Fire TV Stick 4K Max also running Fire OS 7 and the mouse worked as expected while streaming.

**Comment by DaShermanator** [comment]
When you ran your test did you attempt playing any FPS games that utilize relative mouse motion?

**Comment by DaShermanator** [comment]
After nearly a year that I have had this issue it seems that the latest windows update for my host machine (i7-4770k) has resolved my problem. I can't explain it but the mouse is now working as expected with relative mouse motion. I just wanted to say thank you for the amazing open source software and the fact that you even responded to my issue in the first place is amazing. The community really appreciates what you have made. Thank you.

**Comment by nilsheidorn** [comment]
Hi!
I hope its okay to "respawn" this thread.
I am using Moonlight 12.1 on Amazon 4k Gen1 and latest Duo release (that contains a pretty new sunlight) on a AMD Ryzen based "mini gamer" PC.


I seem to have 2 Problems that are similar if not identical to the original post:
Especially visible in Minecraft (Java or Bedrock) running on Host PC:

(1) Mouse behaves normal in Minecraft Menu / options
(2) Mouse is SUPER SENSITIVE in Game (where it controls 3D look direction). Even if sensitivy set to 1% its nearly unplayable (This behavior is Minecraft specific it seems)
(3) Mouse movements have a "border" (easy to see if enabling local cursor display): So if the mouse reaches top/bottom/left/right screen border it stops. THis is the same for e.g. MineCraftor fortnite. And awkward.

Streaming the very same games with Moonlight-on-Steamlink works just fine for buth issues (2) & (3).

Any help / info would be appreciated :-)

**Comment by nilsheidorn** [comment]
Uh oh, MAYBE answering myself, could this behavior be because of this?
-----------------
For non-SHIELD devices and devices running Android 7.1 (Nougat) or earlier, using an external mouse with proper mouse capture on Android requires a rooted device. If you want to use an external mouse on your rooted device, you should download app-root-release.apk from the [GitHub releases page](https://github.com/moonlight-stream/moonlight-android/releases)

**Comment by nilsheidorn** [comment]
Okay, rooted my mantis 4k and installed root-moonlight.
WORKS.

Phewwwww :-)

Hope i do not just look dumb but also help others stranding here :-)
